libs/langgraph/langgraph/pregel/remote.py

```diff
@@ -0,0 +1,360 @@
+from typing import (
+    Any,
+    AsyncIterator,
+    Iterator,
+    Optional,
+    Sequence,
+    Union,
+)
+
+from langchain_core.runnables import RunnableConfig
+from langchain_core.runnables.graph import (
+    Edge as DrawableEdge,
+)
+from langchain_core.runnables.graph import (
+    Graph as DrawableGraph,
+)
+from langchain_core.runnables.graph import (
+    Node as DrawableNode,
+)
+from langgraph_sdk.client import LangGraphClient, SyncLangGraphClient
+from langgraph_sdk.schema import Checkpoint, ThreadState
+from typing_extensions import Self
+
+from langgraph.checkpoint.base import CheckpointMetadata
+from langgraph.pregel.protocol import PregelProtocol
+from langgraph.pregel.types import All, PregelTask, StateSnapshot, StreamMode
+from langgraph.types import Interrupt
+
+
+class RemotePregel(PregelProtocol):
+    def __init__(
+        self, client: LangGraphClient, sync_client: SyncLangGraphClient, graph_id: str
+    ):
+        self.client = client
+        self.sync_client = sync_client
+        self.graph_id = graph_id
+
+    def copy(self, update: dict[str, Any]) -> Self:
+        attrs = {**self.__dict__, **update}
+        return self.__class__(**attrs)
+
+    def with_config(
+        self, config: Optional[RunnableConfig] = None, **kwargs: Any
+    ) -> Self:
+        return self.copy({})
```

Comment:
this seems a bit strange

Reply:
I wasn't sure what to do here because there's no concept of "cloning" a remote Pregel instance (i.e. creating a new Assistant which is a clone of the current one).

To "clone" or "copy" an Assistant, we have to...
1. Call `GET /assistants/<graph_id>`.
1. Update the `configs` (via merge).
1. Call `POST /assistants` to a create a new assistant, the clone/copy (this endpoint doesn't exist right now).

Reply:
I don't think you need to call any endpoints, just save the config in the instance and then use it on every other method?

Reply:
> I don't think you need to call any endpoints, just save the config in the instance and then use it on every other method?

Using `aupdate_state()` as an example... What do we do with `config` passed in the method and `self.config` on the instance. Merge them?

```python
 async def aupdate_state(
    self,
    config: RunnableConfig,
    values: Optional[Union[dict[str, Any], Any]],
    as_node: Optional[str] = None,
) -> RunnableConfig:

    # what do we do with self.config and config?
    self.config

    response: dict = await self.client.threads.update_state(  # type: ignore
        thread_id=config["configurable"]["thread_id"],
        values=values,  # type: ignore
        as_node=as_node,
        checkpoint=self._get_checkpoint(config),
    )
    return self._get_config(response["checkpoint"])
 ```

Reply:
yes exactly merge them. exactly what the real pregel class does